src/components/App/ExpressRecorder.tsx

```diff
@@ -464,10 +495,25 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
                 });
                 return;
             }
+            if (screenOn && screenOn !== shareScreenOn) {
+                navigator.mediaDevices
+                    // @ts-ignore
+                    .getDisplayMedia({ audio: true, video: true })
+                    .then((screenStream: MediaStream) => {
+                        this.setState({ screenStream: screenStream });
+                    })
+                    .catch((e: any) =>
+                        this.handleError("Failed to allocate resource: " + e.message)
+                    );
+            }
             navigator.mediaDevices
                 .getUserMedia(finalConstraints)
                 .then((stream: MediaStream) => {
-                    this.setState({ stream: stream, constraints: finalConstraints });
+                    this.setState({
+                        stream: stream,
+                        constraints: finalConstraints,
+                        shareScreenOn: screenOn
+                    });
```

Comment:
can you explain? why is it an issue? I use different APIs here - getDisplaMedia and getUserMedia

Reply:
normally, you wish to order the dispatching of 2 different promises.
`.getDisplayMedia({ audio: true, video: true })` and `.getUserMedia(finalConstraints)` might be running aside each other with no control or indication of which resolved first/last.
I don't know if you care so much as this might not break the logic yet.
but, I'm saying just to let you know. that's why I said "warning"